tools/sigclient/pkg/query/promqlquery.go

```diff
@@ -0,0 +1,109 @@
+package query
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+	"verifier/pkg/utils"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func RunPromQLQueryFromFile(apiURL string, filepath string) {
+	log.Infof("RunPromQLQueryFromFile: Reading queries from file: %v", filepath)
+	f, err := os.Open(filepath)
+	if err != nil {
+		log.Fatalf("RunPromQLQueryFromFile: Error opening file: %v, err: %v", filepath, err)
+		return
+	}
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	// Skip the first line comment
+	_, err = csvReader.Read()
+	if err != nil {
+		log.Fatalf("RunPromQLQueryFromFile: Error reading file header: %v, err: %v", filepath, err)
+		return
+	}
+
+	for {
+		rec, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatalf("RunPromQLQueryFromFile: Error reading file: %v, err: %v", filepath, err)
+			return
+		}
+
+		if len(rec) != 5 {
+			log.Fatalf("RunPromQLQueryFromFile: Invalid number of columns in query file: [%v]. Expected 5", rec)
+			return
+		}
+
+		query, step := rec[0], rec[3]
+
+		nowTs := uint64(time.Now().Unix())
+		defValue := nowTs - 60*60 // default value is current time - 1 hour
+
+		start := utils.ConvertStringToEpochSec(nowTs, rec[1], defValue)
+		end := utils.ConvertStringToEpochSec(nowTs, rec[2], defValue)
```

Comment:
if end is not given then would we still use the same default value for end and start ?